packages/ui/src/SelectField.tsx

```diff
@@ -0,0 +1,145 @@
+import { DataTestProp } from '@hazelcast/helpers'
+import React, { ReactElement, useRef, FocusEvent, InputHTMLAttributes } from 'react'
+import { v4 as uuid } from 'uuid'
+import cn from 'classnames'
+import ReactSelect, { Props as ReactSelectProps, ValueType, IndicatorProps, ActionMeta } from 'react-select'
+import { ChevronDown, X } from 'react-feather'
+import useIsomorphicLayoutEffect from 'react-use/lib/useIsomorphicLayoutEffect'
+
+import { Error, errorId } from './Error'
+import { Label } from './Label'
+import { Help } from './Help'
+import { Icon } from './Icon'
+import { IconButton } from './IconButton'
+
+import styles from './SelectField.module.scss'
+
+const DropdownIndicator = () => <Icon icon={ChevronDown} ariaHidden size="normal" className={styles.chevron} />
+
+// innerProps set event handling
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const ClearIndicator = ({ innerProps }: IndicatorProps<SelectFieldOption<any>>) => {
+  // Visually impaired people will use the keyboard (backspace) to remove the value. We do not want to confuse them by allowing to focus this button.
+  return <IconButton {...innerProps} icon={X} ariaHidden kind="primary" size="normal" className={styles.clear} tabIndex={-1} />
+}
+
+export type SelectFieldOption<V = string> = {
+  label: string
+  value: V
+}
+
+export type SelectFieldCoreStaticProps = {
+  name: string
+  onBlur?: (e: FocusEvent<HTMLElement>) => void
+  error?: string
+}
+export type SelectFieldCoreDynamicProps<V> =
+  | {
+      isClearable: true
+      value: SelectFieldOption<V> | null
+      onChange: (newValue: SelectFieldOption<V> | null) => void
+    }
+  | {
+      isClearable?: false
+      value: SelectFieldOption<V>
+      onChange: (newValue: SelectFieldOption<V>) => void
+    }
+export type SelectFieldExtraProps<V> = {
+  options: SelectFieldOption<V>[]
+  label: string
+  required?: boolean
+  helperText?: string | ReactElement
+  className?: string
+  errorClassName?: string
+  placeholder?: string
+  menuPortalTarget?: 'body' | 'self' | HTMLElement | null
+} & DataTestProp &
+  Pick<InputHTMLAttributes<HTMLElement>, 'autoFocus' | 'disabled' | 'autoComplete'> &
+  Pick<ReactSelectProps, 'isSearchable' | 'menuIsOpen' | 'menuPlacement' | 'noOptionsMessage' | 'inputValue'>
```

Comment:
Nit: We can pick `required` and `placeholder` from `InputHTMLAttributes` as well.

```suggestion
export type SelectFieldExtraProps<V> = {
  options: SelectFieldOption<V>[]
  label: string
  helperText?: string | ReactElement
  className?: string
  errorClassName?: string
  menuPortalTarget?: 'body' | 'self' | HTMLElement | null
} & DataTestProp &
  Pick<InputHTMLAttributes<HTMLElement>, 'autoFocus' | 'disabled' | 'autoComplete' | 'placeholder' | 'required'> &
  Pick<ReactSelectProps, 'isSearchable' | 'menuIsOpen' | 'menuPlacement' | 'noOptionsMessage' | 'inputValue'>
```